### ISSUE TITLE
Server must have web api for health, stats and other data checking

### ISSUE THREAD

**Issue by JayDi85** [magefree/mage#12016]
General topic about public API and related ideas.

Server must have public web api with basic features like:
* server health status;
* active tables/games/users stats;
* active tables list (id, type, name, settings info, etc);
* user stats and ratings;
* other;

Api usage examples:
- web pages and bots with server status check (current workaround require telnet and show only online status, but not a real work status);
- docker health checking and auto-restarts (current auto-restart works for out of memory errors, but not for freeze/broken server);
- stats collection and monitoring for server admins (online, games amount, etc);
- discord bots and shareable game/tourney links:
  * “copy and paste in app window” easy to implement;
  * “open with xmage” require some research and native os implementation;

Tech details:
- rest endpoints with http/https (for https need some research for compatibility, additional libraries and certificates — but it’s not a critical feature);
- works on some configurable port;
- enabled by default, server config must be able to disable that feature and setup endpoint port;
- response in json format;
- standard rest/web server library with routing and auth/security support (for perspective api like admin tools or online chats), easy to use (like spring with controller annotations, but not a spring);
- perspective: socket.io support for realtime updates and game api;

First steps:
- [ ] find good and active rest java library/server, implement health endpoint;

**Comment by JayDi85** [comment]
@xenohedron I don’t use discord bots, so maybe you have some ideas about possible api endpoints and data that can be useful for historical and other discord groups.

**Comment by theelk801** [comment]
a good option for simple discord integrations is [webhooks](https://discord.com/developers/docs/resources/webhook) which don't require building and running an actual bot

**Comment by xenohedron** [comment]
Yeah, I don't know much about actual bots either, agree that webhooks are a good option to consider

**Comment by theelk801** [comment]
discord bots are pretty easy to set up if you do end up needing one, you just need somewhere to host it
